zetaclient/db/db.go

```diff
@@ -0,0 +1,112 @@
+// Package db represents API for database operations.
+package db
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+
+	"github.com/zeta-chain/zetacore/zetaclient/types"
+)
+
+// SqliteInMemory is a special string to use in-memory database.
+// @see https://www.sqlite.org/inmemorydb.html
+const SqliteInMemory = ":memory:"
+
+// read/write/execute for user
+// read/write for group
+const dirCreationMode = 0o750
+
+var (
+	defaultGormConfig = &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Silent),
+	}
+	migrationEntities = []any{
```

Comment:
can we have 3 groups of migration Entities. One for EVM chain, one for Bitcoin and last one for Solana? They need their own tables.

Reply:
https://gorm.io/docs/migration.html

GORM will create all required tables  based on this list. I don't think this is an issue.

Also, using GORM is an overkill in our case, we can consider simplifying zetaclient's persistent store. Also, we need to build the client in a way that allows loss/flushing sqlite db.

Reply:
Not sure if I clarified it well or not. Each chain creates its own database, they just need their own specific data structures in database.

Bitcoin needs two structs: LastBlockSQLType, TransactionResultSQLType, OutboundHashSQLType
Solana needs one struct:   LastTransactionSQLType
EVM needs three structs:  LastBlockSQLType, TransactionSQLType, ReceiptSQLType

The current impl is creating same group of tables for all chains (for example, created receipt table for Bitcoin, and bitcoin doesn't have a concept or receipt). Not sure what's the benefit of it if the chain doesn't need that table (even in future).